### PR TITLE
NOISSUE: Add Group-based Access Control to charts and update dependencies version

### LINKED ISSUE #136: Add Optional Values with Comments in `values.yaml`
### Is your feature request related to a problem? Please describe.

Yes, the problem is that users of our Helm charts may not be aware of certain configuration options that are available in the templates but are not mandatory and thus not listed in the values.yaml file. This can lead to difficulties in customizing the charts, as users might need to dig into the template files to discover configurable parameters. This can be time-consuming and may lead to unintentional misconfigurations.

### Describe the feature you are requesting, as well as the possible use case(s) for it.

I am requesting the addition of optional values to the values.yaml file, where these values are already present in the templates with default settings. These values should be commented out by default in the values.yaml file, with clear comments explaining their purpose and potential use cases.

### Indicate the importance of this feature to you.
It will significantly improve the user experience by making Helm chart configurations more accessible and understandable, reducing the potential for errors, and increasing the efficiency of the deployment process. While it is not critical, it will contribute to a more polished and user-friendly Helm chart repository.

Must-have

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by arvindh123**
**What does this do?**
This pull request (PR) introduces several changes to the Magistrala Helm chart, providing the following functionalities and improvements:

1. **Enhanced Group-based Access Control:** The PR updates the group-based access control for Magistrala components, specifically for users and things. This improvement strengthens the access control mechanism, allowing for more refined and granular permissions management within Magistrala.

2. **Removal of Auth and Keto Components:** The PR removes the auth and keto components from the Helm charts. This indicates a shift in the architecture or functionality of Magistrala, where these components are no longer necessary.

3. **Dependency Version Updates:** The PR updates the versions of dependencies used in the Helm charts.

4. **Support for Latest Kubernetes Version:** The PR modifies the Helm chart to support the latest version of Kubernetes. This enables Magistrala to be deployed using the updated Helm chart on the most recent Kubernetes release.

5. **Improved Configuration Visibility:** The PR resolves the issue where users were unaware of optional configuration settings not listed in the `values.yaml` file. Optional values present in the templates are now added to the `values.yaml` file, commented out by default with clear explanations, improving user accessibility and reducing the likelihood of misconfigurations.

This PR specifically supports Magistrala version 0.14.0 and above. It is recommended to use this updated Helm chart with Magistrala instances running version 0.14.0 or higher for optimal compatibility and functionality.

**Which issue(s) does this PR fix/relate to?**
- Resolves absmach/magistrala#2141
- Resolves #136

**List any changes that modify/break current functionality**
This PR introduces changes to the Helm chart that impact its compatibility with Magistrala versions. Specifically, the modified Helm chart will only be compatible with Magistrala version 0.14.0 and above. Users attempting to run Magistrala versions lower than 0.14.0 with this updated Helm chart will encounter issues and will not be able to successfully deploy or use it.

**Have you included tests for your changes?**
- Yes

**Did you document any new/modified functionality?**
- Yes, updated `Readme.md`

**Notes**

### CODE REVIEW

**Review by dborovcanin — CHANGES_REQUESTED** [review]
Please fill out the template correctly and resolve conflicts.

**Comment by kibahop** [comment]
Could you please create a contributing guide for **this** repo instead of pointing to another repo. Also my [PR](https://github.com/mainflux/devops/pull/125) was closed in favour of this PR, so do you expect me to continue with this PR? The helm chart should work reliably with current (1.25+) Kubernetes, or do you disagree with this?

**Comment by arvindh123** [comment]
> There is probably no need for new guidelines, but we should update the README with a link to the existing one. I think no actions from you, @arvindh123 included your work in this PR. Right, @arvindh123?

yes @dborovcanin, Included @kibahop changes in this PR 
And  it ready for review

File: charts/mainflux/templates/_helpers.tpl [hunk 1]
```diff
@@ -18,3 +18,11 @@ SPDX-License-Identifier: Apache-2.0
...
 {{- end -}}
 {{- end -}}
+
+{{- define "mainflux.gen-certs" -}}
+{{- $altNames := list ( printf "%s.%s" (include "mainflux.name" .) .Release.Namespace ) ( printf "%s.%s.svc" (include "mainflux.name" .) .Release.Namespace ) -}}
+{{- $ca := genCA "mainflux-ca" 365 -}}
+{{- $cert := genSignedCert ( include "mainflux.name" . ) nil $altNames 365 $ca -}}
+tls.crt: {{ $cert.Cert | b64enc }}
+tls.key: {{ $cert.Key | b64enc }}
+{{- end -}}
```
Comment (dborovcanin) [inline charts/mainflux/templates/_helpers.tpl:28]: Add an empty line.

File: charts/mainflux/templates/ingress.yaml [hunk 2]
```diff
@@ -225,12 +225,12 @@ data:
...
+  namespace: default
 ---
 apiVersion: v1
 kind: ConfigMap
 data:
   5683: "{{ .Release.Namespace }}/{{ .Release.Name }}-adapter-coap:5683"
 metadata:
   name: udp-services
-  namespace: ingress-nginx
+  namespace: default
```
Comment (dborovcanin) [inline charts/mainflux/templates/ingress.yaml:236]: Same.

File: charts/mainflux/values.yaml [hunk 3]
```diff
@@ -352,67 +347,4 @@ nginx_internal:
...
-    requests:
-      cpu: 25m
-  persistence:
-    size: 1Gi
-  commonAnnotations:
-    helm.sh/hook-weight: "-1"
-    helm.sh/hook: "pre-install, pre-upgrade"
-    helm.sh/hook-delete-policy: "before-hook-creation"
-
+    tag: "1.19.1-alpine"
```
Comment (dborovcanin) [inline charts/mainflux/values.yaml:350]: Same.

File: charts/mainflux/values.yaml [hunk 4]
```diff
@@ -7,12 +7,12 @@ defaults:
     pullPolicy: "IfNotPresent"
     repository: "mainflux"
     tag: "latest"
-    imagePullSecrets:
-      - {}
+    # imagePullSecrets:
+    #   - {}
```
Comment (dborovcanin) [inline charts/mainflux/values.yaml:11]: Remove commented code.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

**Review by dborovcanin — COMMENTED** [review]
@JeffMboya Please review this PR.

**Comment by JeffMboya** [comment]
> @JeffMboya Please review this PR.

LGTM

File: charts/mainflux/templates/timescale-reader-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,80 @@
+# Copyright (c) Mainflux
```
Comment (JeffMboya) [inline charts/mainflux/templates/timescale-reader-deployment.yaml:1]:
```suggestion
# Copyright (c) Abstract Machines
```

File: charts/mainflux/templates/timescale-writer-deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,99 @@
+  # Copyright (c) Mainflux
```
Comment (JeffMboya) [inline charts/mainflux/templates/timescale-writer-deployment.yaml:1]:
```suggestion
# Copyright (c) Abstract Machines
```

**Review by JeffMboya — CHANGES_REQUESTED** [review]

File: charts/mainflux/values.yaml [hunk 7]
```diff
@@ -95,324 +181,433 @@ mqtt:
...
+auth:
+  # logLevel: error
+  image: {}
+    # pullSecrets: {}
+    # rootRepository: "magistrala/auth"
+    # tag: "latest"
+    # pullPolicy: "IfNotPresent"
+  # Log level for the auth service. Common options are "debug", "info", "warn", "error".
+  # jaegerTraceRatio: 10
+  # sendTelemetry: true
```
[the remark this answers is not in the document]
Comment (JeffMboya) [reply charts/mainflux/values.yaml:239]: > Why do we have there and some other values commented? It results in errors such as `nil pointer evaluating interface {}.pullSecrets`?

@dborovcanin this is in reference to this issue: [https://github.com/absmach/devops/issues/136](url)

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**